src/imageCache.js

```diff
@@ -170,46 +213,34 @@ export function processExternalGraphicSymbolizers(
   for (let k = 0; k < rules.length; k += 1) {
     const rule = rules[k];
 
-    let symbolizer;
-    let exgraphic;
+    const allSymbolizers = getAllSymbolizers(rule);
+    allSymbolizers.forEach(symbolizer => {
+      externalGraphicPaths.forEach(path => {
```

Comment:
Many nested closures does not increase readability. Maybe some functions can be moved outside the loop.

Reply:
I'm afraid this degree of nesting is necessary. For each rule, all symbolizers have to be examined. For each symbolizer, all possible image locations (externalGraphicPaths) have to be checked to see if the symbolizer contains an image, and if that image's load state has changed.